pkg/cloud/powervs.go

```diff
@@ -88,95 +93,35 @@ type PVMImage struct {
-	} else {
-		bluemixToken = config.IAMAccessToken
-	}
-
-	token, err := jwt.Parse(bluemixToken, func(token *jwt.Token) (interface{}, error) {
-		return "", nil
-	})
-	if err != nil && !strings.Contains(err.Error(), "key is of invalid type") {
-		return &user, err
-	}
-
-	claims := token.Claims.(jwt.MapClaims)
-	if email, ok := claims["email"]; ok {
-		user.Email = email.(string)
-	}
-	user.ID = claims["id"].(string)
-	user.Account = claims["account"].(map[string]interface{})["bss"].(string)
-	iss := claims["iss"].(string)
-	if strings.Contains(iss, "https://iam.cloud.ibm.com") {
-		user.cloudName = "bluemix"
-	} else {
-		user.cloudName = "staging"
-	}
-	user.cloudType = "public"
-
-	user.generation = generation
-	return &user, nil
-}
-
-func NewPowerVSCloud(cloudInstanceID string, debug bool) (Cloud, error) {
-	return newPowerVSCloud(cloudInstanceID, debug)
+func NewPowerVSCloud(cloudInstanceID, zone string, debug bool) (Cloud, error) {
+	return newPowerVSCloud(cloudInstanceID, zone, debug)
 }
 
-func newPowerVSCloud(cloudInstanceID string, debug bool) (Cloud, error) {
+func newPowerVSCloud(cloudInstanceID, zone string, debug bool) (Cloud, error) {
 	apikey := os.Getenv("IBMCLOUD_API_KEY")
-	bxSess, err := bxsession.New(&bluemix.Config{BluemixAPIKey: apikey})
-	if err != nil {
-		return nil, err
-	}
-
-	err = authenticateAPIKey(bxSess)
-	if err != nil {
-		return nil, err
-	}
 
-	user, err := fetchUserDetails(bxSess, 2)
-	if err != nil {
-		return nil, err
+	serviceClientOptions := &resourcecontrollerv2.ResourceControllerV2Options{
+		Authenticator: &core.IamAuthenticator{ApiKey: apikey},
 	}
-
-	ctrlv2, err := controllerv2.New(bxSess)
+	serviceClient, err := resourcecontrollerv2.NewResourceControllerV2UsingExternalConfig(serviceClientOptions)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("errored while creating NewResourceControllerV2UsingExternalConfig: %v", err)
 	}
-
-	resourceClient := ctrlv2.ResourceServiceInstanceV2()
-	in, err := resourceClient.GetInstance(cloudInstanceID)
+	list, _, err := serviceClient.ListResourceInstances(&resourcecontrollerv2.ListResourceInstancesOptions{
+		GUID:           &cloudInstanceID,
+		ResourceID:     &powerVSServiceID,
+		ResourcePlanID: &powerVSPlanID,
+	})
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("errored while listing the Power VS service instance with ID: %s, err: %v", cloudInstanceID, err)
 	}
 
-	zone := in.RegionID
-	region, err := getRegion(zone)
-	if err != nil {
-		return nil, err
+	if len(list.Resources) == 0 {
+		return nil, fmt.Errorf("no Power VS service instance found with ID: %s", cloudInstanceID)
 	}
 
 	authenticator := &core.IamAuthenticator{ApiKey: apikey}
-	piOptions := ibmpisession.IBMPIOptions{Authenticator: authenticator, Debug: debug, Region: region, UserAccount: user.Account, Zone: zone}
+	piOptions := ibmpisession.IBMPIOptions{Authenticator: authenticator, Debug: debug, UserAccount: *list.Resources[0].AccountID, Zone: zone}
```

Comment:
can we make it more meaningful like below:
```suggestion
	piOptions := ibmpisession.IBMPIOptions{Authenticator: authenticator, Debug: debug, UserAccount: *resourceInstanceList.Resources[0].AccountID, Zone: zone}
```